### ISSUE TITLE
False positive when secondary variable non-nullability infers first variable non-nullability

### ISSUE THREAD

**Issue by unv-unv** [uber/NullAway#1801]
```java
class Test {
    Object foo(@Nullable Object param) {
        Object check = param == null ? null : param;
        if (check != null) {
            // BUG: returning @Nullable expression from method with @NonNull return type
            return param;
        }
        return new Object();
    }
}
```

Shows compilation error:
```
[ERROR] Test.java:[6,12] [NullAway] returning @Nullable expression from method with @NonNull return type
[ERROR]     (see http://t.uber.com/nullaway )
```

NullAway 0.14.1, Error Prone 2.50.0.

**Comment by msridhar** [comment]
This is a known limitation of NullAway; see #98.  We've thought for a while about implementing more precise dataflow analysis for certain cases, but we have higher priority issues to fix for the moment.  Hopefully eventually we will get to it.  Sorry for the false positive.
